os/drivers/lcd/st7789.c

```diff
@@ -0,0 +1,920 @@
+	vinfo->xres = ST7789_XRES;	/* Horizontal resolution in pixel columns */
+	vinfo->yres = ST7789_YRES;	/* Vertical resolution in pixel rows */
+	vinfo->nplanes = 1;			/* Number of color planes supported */
+	return OK;
+}
+
+/****************************************************************************
+ * Name:  st7789_getplaneinfo
+ *
+ * Description:
+ *   Get information about the configuration of each LCD color plane.
+ *
+ ****************************************************************************/
+
+static int st7789_getplaneinfo(FAR struct lcd_dev_s *dev, unsigned int planeno, FAR struct lcd_planeinfo_s *pinfo)
+{
+	FAR struct st7789_dev_s *priv = (FAR struct st7789_dev_s *)dev;
+	DEBUGASSERT(dev && pinfo && planeno == 0);
+	pinfo->putrun = st7789_putrun;	/* Put a run into LCD memory */
+	pinfo->putarea = st7789_putarea;	/* Put an area into LCD */
+#ifndef CONFIG_LCD_NOGETRUN
+	pinfo->getrun = st7789_getrun;	/* Get a run from LCD memory */
+#endif
+	pinfo->buffer = (FAR uint8_t *) priv->runbuffer;	/* Run scratch buffer */
+	pinfo->bpp = priv->bpp;		/* Bits-per-pixel */
+	pinfo->dev = dev;			/* The lcd device */
+	return OK;
+}
+
+/****************************************************************************
+ * Name:  st7789_getpower
+ ****************************************************************************/
+
+static int st7789_getpower(FAR struct lcd_dev_s *dev, int color)
+{
+	FAR struct st7789_dev_s *priv = (FAR struct st7789_dev_s *)dev;
+	return priv->power;
+}
+
+/****************************************************************************
+ * Name:  st7789_setpower
+ ****************************************************************************/
+
+static int st7789_setpower(FAR struct lcd_dev_s *dev, int power)
+{
+	FAR struct st7789_dev_s *priv = (FAR struct st7789_dev_s *)dev;
+	DEBUGASSERT((unsigned)power <= CONFIG_LCD_MAXPOWER);
+	/* Set new power level */
+	lcd_lock(priv->spi);
+	if (power > 0) {
+		st7789_sendcmd(priv, ST7789_DISPON);
+		priv->power = power;
+	} else {
+		st7789_sendcmd(priv, ST7789_DISPOFF);
+		priv->power = 0;
+	}
+	lcd_unlock(priv->spi);
+	return OK;
+}
+
+/****************************************************************************
+ * Name:  st7789_init
+ *
+ * Description:
+ *   init the lcd
+ *
+ ****************************************************************************/
+
+static int st7789_init(FAR struct lcd_dev_s *dev)
+{
+	FAR struct st7789_dev_s *priv = (FAR struct st7789_dev_s *)dev;
+	lcd_lock(priv->spi);
+	st7789_init_cmd(priv);
+	st7789_bpp(priv, ST7789_BPP);
+	st7789_sendcmd(priv, ST7789_DISPON);
+	lcd_unlock(priv->spi);
+	return OK;
+}
+
+/****************************************************************************
+ * Name:  st7789_clear
+ *
+ * Description:
+ *   clear the lcd/ fill with black color
+ *
+ ****************************************************************************/
+
+static int st7789_clear(FAR struct lcd_dev_s *dev)
```

Comment:
I think this is unnecessary. user can make block screen with 'putrun', If I am wrong, please let me know.